cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/spf13/cobra"
+
+	"github.com/ava-labs/apm/types"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	alias          string
+	repoURL        string
+	vmDescPath     string
+	subnetDescPath string
+)
+
+type newPublisherFunc func(string, string, string) subnet.Publisher
+
+// avalanche subnet deploy
+func newPublishCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "publish [subnetName]",
+		Short:        "Publish the subnet's VM to a repository",
+		Long:         ``,
+		SilenceUsage: true,
+		RunE:         publish,
+		Args:         cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&alias, "alias", "", "An alias for the repo to publish to")
```

Comment:
Reading the help string alone, I'm not sure what this is supposed to be. Maybe we need an example value

Reply:
Trying this time with `We publish to a remote repo, but store identify the repo locally under a user-provided alias (e.g. myrepo).`